src/pages/components/BookmarkItem.qml

```diff
@@ -0,0 +1,119 @@
+/****************************************************************************
+**
+** Copyright (C) 2020 Open Mobile Platform LLC
+**
+****************************************************************************/
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import QtQuick 2.0
+import Sailfish.Silica 1.0
+import Sailfish.Browser 1.0
+
+ListItem {
+    id: root
+
+    function openLink() {
+        webView.tabModel.newTab(model.url)
+        pageStack.pop()
+    }
+
+    contentHeight: Math.max(Theme.itemSizeMedium, column.height + 2*Theme.paddingMedium)
+
+    menu: contextMenuComponent
+
+    ListView.onAdd: AddAnimation { target: root }
+
+    FavoriteIcon {
+        id: favoriteIcon
+
+        anchors {
+            verticalCenter: parent.verticalCenter
+            left: parent.left
+            leftMargin: Theme.horizontalPageMargin
+        }
+
+        icon: model.favicon
+
+        sourceSize.width: Theme.iconSizeMedium
+        sourceSize.height: Theme.iconSizeMedium
+        width: Theme.iconSizeMedium
+        height: Theme.iconSizeMedium
+    }
+
+    Column {
+        id: column
+        anchors {
+            verticalCenter: parent.verticalCenter
+            left: favoriteIcon.right
+            leftMargin: Theme.paddingMedium
+            right: parent.right
+            rightMargin: Theme.horizontalPageMargin
+        }
+
+        Label {
+            text: Theme.highlightText(url, searchText, Theme.highlightColor)
+            textFormat: Text.StyledText
+            opacity: Theme.opacityHigh
```

Comment:
This looks odd to me as the "primary" text line is now a bit dimmed and so called "description label" has the primary text's color. Let's have it like this for now and I'll sync with design still.

Thanks @atatarov !